### PR TITLE
Add command methods

### PR DESCRIPTION
**Pull request by jballoffet**
# 🎉 New feature

Related: #87

## Summary

This PR adds callback methods for each CLI command as a stepping stone towards decoupling the main application logic from the shell and CLI business logic. It also removes some commands that may result in HW damage if wrongly used.

More PRs will be sent to keep improving the code structure.

Note: this PR is expected to be submitted after #186.

## Test it

Tested by performing manual integration tests on the robot hardware.

## Checklist
- [X] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if it affects the public API)

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: andino_firmware/src/commands.h [hunk 1]
```diff
@@ -62,24 +62,26 @@
...
+  /// @brief Reads the encoders tick count values.
+  static constexpr char kreadEncoders{'e'};
+  /// @brief Sets the encoders ticks count to zero.
+  static constexpr char kresetEncoders{'r'};
+  /// @brief Sets the motors speed [ticks/s].
+  static constexpr char ksetMotorsSpeed{'m'};
+  /// @brief Sets the motors PWM value [duty range: 0-255].
+  static constexpr char ksetMotorsPwm{'o'};
+  /// @brief Sets the PIDs tuning gains [format: "kp:kd:ki:ko"].
+  static constexpr char ksetPidsTuningGains{'u'};
```
Comment (francocipollone) [inline andino_firmware/src/commands.h:84]: nit: `set` should be `Set`

File: andino_firmware/src/app.cpp [hunk 2]
```diff
@@ -199,114 +195,142 @@ void App::reset_command() {
...
     default:
-      Serial.println("Invalid Command");
+      cmd_unknown(argv1, argv2);
       break;
   }
 }
 
+void App::cmd_unknown(const char* arg1, const char* arg2) {
+  (void)arg1;
+  (void)arg2;
```
Comment (francocipollone) [inline andino_firmware/src/app.cpp:233]: nit: 
1. static_cast can be used here to be more `cpp`
2. I think that simply omitting the parameters hints the compiler that the args won't be used.
```cpp
void App::cmd_unknown(const char* , const char* ) {
```

File: andino_firmware/src/app.cpp [hunk 3]
```diff
@@ -199,114 +195,142 @@ void App::reset_command() {
...
+
+void App::cmd_read_analog_gpio(const char* arg1, const char* arg2) {
+  (void)arg2;
+  int pin = atoi(arg1);
+  Serial.println(analogRead(pin));
+}
+
+void App::cmd_read_digital_gpio(const char* arg1, const char* arg2) {
+  (void)arg2;
+  int pin = atoi(arg1);
```
Comment (francocipollone) [inline andino_firmware/src/app.cpp:245]: nit: use const whenever possible

**Review by francocipollone — CHANGES_REQUESTED** [review]
Thanks improving code structure. I left some comments

File: andino_firmware/src/commands.h (hunk 1 above)
Comment (jballoffet) [reply andino_firmware/src/commands.h]: Good catch, done!

File: andino_firmware/src/app.cpp (hunk 2 above)
Comment (jballoffet) [reply andino_firmware/src/app.cpp]: Done.

File: andino_firmware/src/app.cpp [hunk 4]
```diff
@@ -199,114 +195,142 @@ void App::reset_command() {
...
+  left_encoder_.reset();
+  right_encoder_.reset();
+  left_pid_controller_.reset(left_encoder_.read());
+  right_pid_controller_.reset(right_encoder_.read());
+  Serial.println("OK");
+}
+
+void App::cmd_set_motors_speed(const char* arg1, const char* arg2) {
+  int left_motor_speed = atoi(arg1);
+  int right_motor_speed = atoi(arg2);
```
[the remark this answers is not in the document]
Comment (jballoffet) [reply andino_firmware/src/app.cpp]: Done.

File: andino_firmware/src/app.cpp [hunk 5]
```diff
@@ -199,114 +195,142 @@ void App::reset_command() {
...
+  // The target speeds are in ticks per second, so we need to convert them to ticks per
+  // Constants::kPidRate.
+  left_pid_controller_.set_setpoint(left_motor_speed / Constants::kPidRate);
+  right_pid_controller_.set_setpoint(right_motor_speed / Constants::kPidRate);
+  Serial.println("OK");
+}
+
+void App::cmd_set_motors_pwm(const char* arg1, const char* arg2) {
+  int left_motor_pwm = atoi(arg1);
+  int right_motor_pwm = atoi(arg2);
```
[the remark this answers is not in the document]
Comment (jballoffet) [reply andino_firmware/src/app.cpp]: Done.

**Comment by jballoffet** [comment]
> BTW, is the `o ` command still failing?

Yes, I will fix it in a subsequent PR so as to ease the review process. Thanks for checking!

**Review by francocipollone — APPROVED** [review]
LGTM!

**Pull request merged**